src/GridFS/WritableStream.php

```diff
@@ -173,6 +174,33 @@ public function close(): void
         $this->isClosed = true;
     }
 
+    /**
+     * Delete all files and chunks associated with this stream filename.
+     *
+     * @return int The number of deleted files
+     */
+    public function delete(): int
+    {
+        /** @var Document[] $revisions */
+        $revisions = $this->collectionWrapper->findFiles(['filename' => $this->file['filename']], [
+            'codec' => null,
+            'typeMap' => ['root' => 'bson'],
+            'projection' => ['_id' => 1],
+        ]);
+
+        $count = 0;
+
+        foreach ($revisions as $file) {
+            $this->collectionWrapper->findFileById($file->get('_id'));
+            $this->collectionWrapper->deleteFileAndChunksById($file->get('_id'));
+            $count++;
+        }
```

Comment:
I'm wondering if we can improve on this by collecting all IDs, then deleting all files with the given filename, followed by removing all chunks for the IDs we previously found.

While there's always a chance to end up in an undesirable state, e.g. by having older revisions of files remaining, the suggestion above would reduce this by instructing the server to delete all file documents with a given name, ensuring that future requests for such files would not produce any result. Thus, if there was an error while deleting any chunks, we wouldn't end with older revisions left over.

Reply:
I will implement a solution consistent with `rename` https://github.com/mongodb/mongo-php-library/pull/1207#discussion_r1430167458

Reply:
This would definitely benefit from some bulk strategy to remove the `fs.files` documents first (likely using `$in` on multiple `_id` values), and _then_ cleaning up the orphan chunks. I think it'd also benefit from being addressed in the GridFS spec, since we'd probably want to have a consistent way to address this in any driver that wishes to do so. And codifying that in a spec would ensure we're not overlooking some edge case.